### ISSUE TITLE
Add "after" delimiter-adding support to "paste"

### ISSUE THREAD

**Issue by pokey** [pokey/cursorless-vscode#128]
See https://github.com/pokey/cursorless-vscode/blob/master/src/actions/BringMoveSwap.ts#L111-L121

**Comment by AndreasArvidsson** [comment]
How is this going to work when we use the build in paste command?

**Comment by AndreasArvidsson** [comment]
That can be problematic combined with #127

**Comment by pokey** [comment]
@AndreasArvidsson is this one not solved in latest slew of PRs?

**Comment by AndreasArvidsson** [comment]
@pokey Not at all. Since we are using the built in paste now it would require large rewrites to get it working.

**Comment by pokey** [comment]
Ah right fair point. I really go back and forth on the built-in paste vs custom. Another argument for custom is that I think I'd argue that "bring" and "paste" should behave roughly the same. But yeah then we'd need to recreate the multi-line multi-selection copy tracking logic. Idk

**Comment by pokey** [comment]
Although couldn't we solve this one fairly easily by inserting delimiter, then moving cursor before / after it then issuing paste?

**Comment by AndreasArvidsson** [comment]
Yes that's the only viable solution but that's not easy right now with how the code looks. Then we will have one edit we control and one edit we don't and we need to update selections on both. Doable of course but not that easy.

**Comment by AndreasArvidsson** [comment]
I think the best way to solve this is
* Revert to our own implementation of copy/cut/paste
* Try to leverage as much from bring/move as we can
* Use the same implementation that vscode does with an internal clipboard per cursor copied
* Support clipboard history. 
* Contribute our own ctrl/cmd+c/x/v keyboard bindings with a when clause that the editor is focused

@pokey Sounds good?

**Comment by pokey** [comment]
Looks good to me! Worth mentioning that we should make sure clipboard state is kept in global storage so it can be shared across windows

Also would be awesome if you're able to scare up that link to the VSCode implementation again

**Comment by AndreasArvidsson** [comment]
Some of the interesting parts are here

https://github.com/microsoft/vscode/blob/1498d0f34053f854e75e1364adaca6f99e43de08/src/vs/editor/contrib/clipboard/browser/clipboard.ts#L230

https://github.com/microsoft/vscode/blob/1fa4c98b130a9c70d4baeb59979db0873bee93ed/src/vs/editor/common/cursor/cursorTypeOperations.ts#L109

I can't right now find where they actually check that you haven't updated the clipboard outside of vscode. But I would assume it's something like:
```
if (clipboard.text === multiCursor.expectedText) { .. }
```

And if we want a clipboard history we definitely need to do something like this.

**Comment by pokey** [comment]
I was never able to find that check either fwiw.  Slightly worrying 🤔

**Comment by AndreasArvidsson** [comment]
According to my tests they must have a check like that or a hook whenever the clipboard changes because if you copy something out of vscode the paste behavior/content is updated.

**Comment by pokey** [comment]
I mean yeah I can definitely believe it, just slightly worrying neither of us can find the code. Would be good if we understand their code in order to make sure there isn't some fancy corner case handling that we miss by rolling our own
